dependencies/FindFastJet.cmake

```diff
@@ -68,28 +68,35 @@ if(${PKGNAME}_CONFIG)
     if ("CGAL" IN_LIST ${PKGNAME}_LIBS)
       find_library(lib_cgal NAMES "CGAL" PATHS $ENV{CGAL_ROOT}/lib NO_DEFAULT_PATH)
       if (NOT lib_cgal)
-        message(FATAL_ERROR "CGAL not found in $ENV{CGAL_ROOT}/lib")
+        if(FastJet_FIND_REQUIRED)
```

Comment:
This part is only reached if `fastjet-config` is found, and in this case there is a problem if it reports the use of CGAL but we cannot find the library.

Reply:
I guess @matthiasrichter goal here was to prevent a plain crash of the CMake configure stage in cases where find_package(FastJet) is called without the REQUIRED argument ? The fact that CGAL is not found (where it should be) is indeed an issue, but should only be fatal one if FastJet is required.

Reply:
@aphecetche I disagree, finding fastjet exposing linkage against CGAL but not finding CGAL itself would result in an inconsistent usage. This should not happen and, for me, such an inconsistency (package exposing another library through `<pkg>-config`, the latter one not found) warrants a failure of the configuration. 

If we deliberately decide to handle this gracefully (even though I don't see a good reason to do so), we would have to invalidate the finding of FastJet itself and treat also FastJet as not found. 

@matthiasrichter In order to better understand what you were after here, did you actually observe a failure related to this? If so, under what conditions?

Reply:
> If we deliberately decide to handle this gracefully (even though I don't see a good reason to do so), we would have to invalidate the finding of FastJet itself and treat also FastJet as not found.

Sure. If anything goes wrong in the FindFastJet.cmake, then FastJet must be declared NOT FOUND.